plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -107,6 +120,74 @@ export class PgInstrumentation extends InstrumentationBase {
     return [modulePG, modulePGPool];
   }
 
+  private _getClientConnectPatch() {
+    const plugin = this;
+    return (original: typeof pgTypes.Client.prototype.connect) => {
+      return function connect(
+        this: pgTypes.Client,
+        callback?: PgErrorCallback
+      ) {
+        const span = plugin.tracer.startSpan(
+          `${PgInstrumentation.COMPONENT}.connect`,
+          {
+            kind: SpanKind.CLIENT,
+            attributes: {
+              [SemanticAttributes.DB_SYSTEM]: DbSystemValues.POSTGRESQL,
+              [SemanticAttributes.DB_NAME]: this.database,
+              [SemanticAttributes.NET_PEER_NAME]: this.host,
+              [SemanticAttributes.DB_CONNECTION_STRING]:
+                utils.getConnectionString(this),
+              [SemanticAttributes.NET_PEER_PORT]: this.port,
+              [SemanticAttributes.DB_USER]: this.user,
+            },
+          }
+        );
+
+        if (callback) {
+          const parentSpan = trace.getSpan(context.active());
+          callback = utils.patchConnectErrorCallback(span, callback);
+          if (parentSpan) {
+            callback = context.bind(context.active(), callback);
+          }
+        }
+
+        const connectResult: unknown = context.with(
+          trace.setSpan(context.active(), span),
+          () => {
+            return original.call(this, callback as never);
+          }
+        );
+
+        if (!(connectResult instanceof Promise)) {
+          return connectResult;
+        }
+
+        const connectResultPromise = connectResult as Promise<unknown>;
+        return context.bind(
+          context.active(),
+          connectResultPromise
+            .then(
+              result =>
+                new Promise(resolve => {
+                  span.end();
+                  resolve(result);
+                })
+            )
+            .catch((error: Error) => {
+              return new Promise((_, reject) => {
+                span.setStatus({
+                  code: SpanStatusCode.ERROR,
+                  message: error.message,
+                });
+                span.end();
+                reject(error);
+              });
+            })
+        );
+      };
+    };
```

Comment:
This block is a duplicate of the code in `_getPoolConnectPatch`.
Could be nice to extract into a function and invoke from both places

Reply:
Refactored this into a single function named `handleConnectResult`